### PR TITLE
Update `User` model comments about permissions

### LINKED ISSUE #17467: Do we have some documents do describe our permission system? I have questions about the real meaning about "prohibited", "active", "restricted" and how they work and how they affect the permission checks.
Do we have some documents do describe our permission system? I have questions about the real meaning about "prohibited", "active", "restricted" and how they work and how they affect the permission checks.

_Originally posted by @wxiaoguang in https://github.com/go-gitea/gitea/issues/17464#issuecomment-953470628_

### PR DESCRIPTION
**Pull request by wxiaoguang**
Copied from and close https://github.com/go-gitea/gitea/issues/17467

Please help to check whether my understanding and comments are correct.

### CODE REVIEW

File: models/user.go [hunk 1]
```diff
@@ -131,14 +131,21 @@ type User struct {
 	// Maximum repository creation limit, -1 means use global default
 	MaxRepoCreation int `xorm:"NOT NULL DEFAULT -1"`
 
-	// Permissions
-	IsActive                bool `xorm:"INDEX"` // Activate primary email
-	IsAdmin                 bool
-	IsRestricted            bool `xorm:"NOT NULL DEFAULT false"`
+	// IsActive true: primary email is activated, user can access Web UI and Git SSH.
```
Comment (Gusted) [inline models/user.go:134]: `activated` can be vague, I think `verified` can be used instead.
```suggestion
	// IsActive true: primary email is verified, user can access the Web UI and Git SSH.
```

File: models/user.go [hunk 2]
```diff
@@ -131,14 +131,21 @@ type User struct {
...
+	// false: an inactive user can only log in Web UI for account operations (ex: activate the account by email), no other access.
```
Comment (Gusted) [inline models/user.go:135]:
```suggestion
	// false: an non-verified user, who can only log in Web UI for account operations
	// (ex: activate the account by email), and don't have any other access.
```

File: models/user.go [hunk 3]
```diff
@@ -131,14 +131,21 @@ type User struct {
...
+	IsActive bool `xorm:"INDEX"`
+	// user is a Gitea admin, can access all repositories and admin pages
```
Comment (Gusted) [inline models/user.go]:
```suggestion
	// user is a Gitea admin, who can access all repositories and the admin pages.
```

File: models/user.go [hunk 4]
```diff
@@ -131,14 +131,21 @@ type User struct {
...
+	IsAdmin bool
+	// true: user is allowed to see only organizations/repositories that he has explicit rights specified
```
Comment (Gusted) [inline models/user.go]:
```suggestion
	// true: user is only allowed to see organizations/repositories that they has explicit rights to.
```

File: models/user.go [hunk 5]
```diff
@@ -131,14 +131,21 @@ type User struct {
...
+	// (ex: in private Gitea instances user won't be allowed to see even organizations/repositories that are set as public)
+	IsRestricted bool `xorm:"NOT NULL DEFAULT false"`
+
 	AllowGitHook            bool
 	AllowImportLocal        bool // Allow migrate repository by local path
 	AllowCreateOrganization bool `xorm:"DEFAULT true"`
-	ProhibitLogin           bool `xorm:"NOT NULL DEFAULT false"`
+
+	// true: user is not allowed to log in Web UI. Git SSH access could still be allowed.
```
Comment (Gusted) [inline models/user.go]:
```suggestion
	// true: user is not allowed to log into the Web UI. Git SSH access could still be allowed.
```

I think ` Git SSH access >could< still be allowed.` that `could` could be clarified, as in, which conditions the Git SSH access is allowed.

**Review by Gusted — CHANGES_REQUESTED** [review]
The descriptions are good(if I've read the code correctly) and describes what the fields are, I think the wording need a little bit of change and maybe clarify some things up.

File: models/user.go (hunk 1 above)
Comment (wxiaoguang) [reply models/user.go:134]: We should keep "activated", this concept is used widely in Gitea code base.

File: models/user.go (hunk 2 above)
Comment (wxiaoguang) [reply models/user.go:135]: As above. We do not need to introduce new concepts. Just follow the old ones.

File: models/user.go (hunk 5 above)
Comment (wxiaoguang) [reply models/user.go]: The conditions belong to SSH access module, User model has nothing to do (no knowledge) about it. 

If we need to clarify the conditions, related comments should be written in SSH access module.

File: models/user.go (hunk 4 above)
Comment (wxiaoguang) [reply models/user.go]: Still a syntax error "they has", so I use "he/she has" now. 😊

Comment (wxiaoguang) [reply models/user.go]: Oh sorry, I am not a native speaker, just realized that `singular they` is correct for gender-neutral language (I googled it from Wikipedia)

File: models/user.go (hunk 1 above)
Comment (Gusted) [reply models/user.go:134]: That makes sense.

File: models/user.go (hunk 5 above)
Comment (Gusted) [reply models/user.go]: I dont have a problem if its clarified to the SSH access module. 

For someone who's trying understand what a field does and it refers to something that isnt described somewhere, then I would personally still not fully understand what the field does.

**Review by Gusted — APPROVED** [review]
The `prohibtedLogin` could be done in another PR as this is just for the user struct. LGTM

File: models/user.go (hunk 5 above)
Comment (wxiaoguang) [reply models/user.go]: Got it. I just added a new comment: "please refer to Git/SSH access related code/documents".

**Review by lunny — APPROVED** [review]

**Comment by wxiaoguang** [comment]
trigger bot

**Pull request merged**